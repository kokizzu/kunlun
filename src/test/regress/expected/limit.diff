120,121c120,121
<                                                          QUERY PLAN                                                         
< ----------------------------------------------------------------------------------------------------------------------------
---
>                            QUERY PLAN                           
> ----------------------------------------------------------------
123,129c123,126
<    Output: unique1, unique2, ("nextval"('testseq'::regclass))
<    ->  Result
<          Output: unique1, unique2, "nextval"('testseq'::regclass)
<          ->  RemoteScan on public.tenk1
<                Output: unique1, unique2
<                Shard: 2  Remote SQL: select  tenk1.unique1, tenk1.unique2 from regress_$$_public.tenk1 tenk1 ORDER BY 2 ASC
< (7 rows)
---
>    Output: unique1, unique2, (nextval('testseq'::regclass))
>    ->  Index Scan using tenk1_unique2 on public.tenk1
>          Output: unique1, unique2, nextval('testseq'::regclass)
> (4 rows)
142,143c139,140
<                                                                  QUERY PLAN                                                                 
< --------------------------------------------------------------------------------------------------------------------------------------------
---
>                                 QUERY PLAN                                
> --------------------------------------------------------------------------
145c142
<    Output: unique1, unique2, ("nextval"('testseq'::regclass)), tenthous
---
>    Output: unique1, unique2, (nextval('testseq'::regclass)), tenthous
147,148c144,145
<          Output: unique1, unique2, "nextval"('testseq'::regclass), tenthous
<          ->  RemoteScan on public.tenk1
---
>          Output: unique1, unique2, nextval('testseq'::regclass), tenthous
>          ->  Sort
150,151c147,150
<                Shard: 2  Remote SQL: select  tenk1.unique1, tenk1.unique2, tenk1.tenthous from regress_$$_public.tenk1 tenk1 ORDER BY 3 ASC
< (7 rows)
---
>                Sort Key: tenk1.tenthous
>                ->  Seq Scan on public.tenk1
>                      Output: unique1, unique2, tenthous
> (9 rows)
164,165c163,164
<                                                          QUERY PLAN                                                         
< ----------------------------------------------------------------------------------------------------------------------------
---
>                                                                          QUERY PLAN                                                                          
> -------------------------------------------------------------------------------------------------------------------------------------------------------------
170,173c169,171
<          ->  RemoteScan on public.tenk1
<                Output: unique1, unique2
<                Shard: 2  Remote SQL: select  tenk1.unique1, tenk1.unique2 from regress_$$_public.tenk1 tenk1 ORDER BY 2 ASC
< (7 rows)
---
>          ->  Index Scan using tenk1_unique2 on public.tenk1
>                Output: unique1, unique2, two, four, ten, twenty, hundred, thousand, twothousand, fivethous, tenthous, odd, even, stringu1, stringu2, string4
> (6 rows)
184,185c182,183
<                                                                  QUERY PLAN                                                                 
< --------------------------------------------------------------------------------------------------------------------------------------------
---
>                              QUERY PLAN                             
> --------------------------------------------------------------------
190c188
<          ->  RemoteScan on public.tenk1
---
>          ->  Sort
192,193c190,193
<                Shard: 2  Remote SQL: select  tenk1.unique1, tenk1.unique2, tenk1.tenthous from regress_$$_public.tenk1 tenk1 ORDER BY 3 ASC
< (7 rows)
---
>                Sort Key: tenk1.tenthous
>                ->  Seq Scan on public.tenk1
>                      Output: unique1, unique2, tenthous
> (9 rows)
245,246c245,246
<                                                                         QUERY PLAN                                                                         
< -----------------------------------------------------------------------------------------------------------------------------------------------------------
---
>                                                     QUERY PLAN                                                     
> -------------------------------------------------------------------------------------------------------------------
248,253c248,253
<    Output: (sum(tenthous)), ((((sum(tenthous)))::double precision + (random() * '0'::double precision))), thousand
<    ->  Result
<          Output: (sum(tenthous)), (((sum(tenthous)))::double precision + (random() * '0'::double precision)), thousand
<          ->  RemoteScan on public.tenk1
<                Output: (sum(tenthous)), thousand
<                Shard: 2  Remote SQL: select  SUM(tenk1.tenthous), tenk1.thousand from regress_$$_public.tenk1 tenk1 GROUP BY tenk1.thousand ORDER BY 2 ASC
---
>    Output: (sum(tenthous)), (((sum(tenthous))::double precision + (random() * '0'::double precision))), thousand
>    ->  GroupAggregate
>          Output: sum(tenthous), ((sum(tenthous))::double precision + (random() * '0'::double precision)), thousand
>          Group Key: tenk1.thousand
>          ->  Index Only Scan using tenk1_thous_tenthous on public.tenk1
>                Output: thousand, tenthous
