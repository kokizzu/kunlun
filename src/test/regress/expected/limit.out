--
-- LIMIT
-- Check the LIMIT/OFFSET feature of SELECT
--
SELECT ''::text AS two, unique1, unique2, stringu1
		FROM onek WHERE unique1 > 50
		ORDER BY unique1 LIMIT 2;
 two | unique1 | unique2 | stringu1 
-----+---------+---------+----------
(0 rows)

SELECT ''::text AS five, unique1, unique2, stringu1
		FROM onek WHERE unique1 > 60
		ORDER BY unique1 LIMIT 5;
 five | unique1 | unique2 | stringu1 
------+---------+---------+----------
(0 rows)

SELECT ''::text AS two, unique1, unique2, stringu1
		FROM onek WHERE unique1 > 60 AND unique1 < 63
		ORDER BY unique1 LIMIT 5;
 two | unique1 | unique2 | stringu1 
-----+---------+---------+----------
(0 rows)

SELECT ''::text AS three, unique1, unique2, stringu1
		FROM onek WHERE unique1 > 100
		ORDER BY unique1 LIMIT 3 OFFSET 20;
 three | unique1 | unique2 | stringu1 
-------+---------+---------+----------
(0 rows)

SELECT ''::text AS zero, unique1, unique2, stringu1
		FROM onek WHERE unique1 < 50
		ORDER BY unique1 DESC LIMIT 8 OFFSET 99;
 zero | unique1 | unique2 | stringu1 
------+---------+---------+----------
(0 rows)

SELECT ''::text AS eleven, unique1, unique2, stringu1
		FROM onek WHERE unique1 < 50
		ORDER BY unique1 DESC LIMIT 20 OFFSET 39;
 eleven | unique1 | unique2 | stringu1 
--------+---------+---------+----------
(0 rows)

SELECT ''::text AS ten, unique1, unique2, stringu1
		FROM onek
		ORDER BY unique1 OFFSET 990;
 ten | unique1 | unique2 | stringu1 
-----+---------+---------+----------
(0 rows)

SELECT ''::text AS five, unique1, unique2, stringu1
		FROM onek
		ORDER BY unique1 OFFSET 990 LIMIT 5;
 five | unique1 | unique2 | stringu1 
------+---------+---------+----------
(0 rows)

SELECT ''::text AS five, unique1, unique2, stringu1
		FROM onek
		ORDER BY unique1 LIMIT 5 OFFSET 900;
 five | unique1 | unique2 | stringu1 
------+---------+---------+----------
(0 rows)

-- Test null limit and offset.  The planner would discard a simple null
-- constant, so to ensure executor is exercised, do this:
select * from int8_tbl limit (case when random() < 0.5 then null::bigint end);
        q1        |        q2         
------------------+-------------------
              123 |               456
              123 |  4567890123456789
 4567890123456789 |               123
 4567890123456789 |  4567890123456789
 4567890123456789 | -4567890123456789
(5 rows)

select * from int8_tbl offset (case when random() < 0.5 then null::bigint end);
        q1        |        q2         
------------------+-------------------
              123 |               456
              123 |  4567890123456789
 4567890123456789 |               123
 4567890123456789 |  4567890123456789
 4567890123456789 | -4567890123456789
(5 rows)

-- Stress test for variable LIMIT in conjunction with bounded-heap sorting
SELECT
  (SELECT n
     FROM (VALUES (1)) AS x,
          (SELECT n FROM generate_series(1,10) AS n
             ORDER BY n LIMIT 1 OFFSET s-1) AS y) AS z
  FROM generate_series(1,10) AS s;
 z  
----
  1
  2
  3
  4
  5
  6
  7
  8
  9
 10
(10 rows)

--
-- Test behavior of volatile and set-returning functions in conjunction
-- with ORDER BY and LIMIT.
--
create temp sequence testseq;
CREATE SEQUENCE
explain (verbose, costs off)
select unique1, unique2, nextval('testseq')
  from tenk1 order by unique2 limit 10;

select unique1, unique2, nextval('testseq')
  from tenk1 order by unique2 limit 10;
 unique1 | unique2 | nextval 
---------+---------+---------
(0 rows)

select currval('testseq');
psql:sql/limit.sql:62: ERROR:  currval of sequence "testseq" is not yet defined in this session
explain (verbose, costs off)
select unique1, unique2, nextval('testseq')
  from tenk1 order by tenthous limit 10;

select unique1, unique2, nextval('testseq')
  from tenk1 order by tenthous limit 10;
 unique1 | unique2 | nextval 
---------+---------+---------
(0 rows)

select currval('testseq');
psql:sql/limit.sql:71: ERROR:  currval of sequence "testseq" is not yet defined in this session
explain (verbose, costs off)
select unique1, unique2, generate_series(1,10)
  from tenk1 order by unique2 limit 7;

select unique1, unique2, generate_series(1,10)
  from tenk1 order by unique2 limit 7;
 unique1 | unique2 | generate_series 
---------+---------+-----------------
(0 rows)

explain (verbose, costs off)
select unique1, unique2, generate_series(1,10)
  from tenk1 order by tenthous limit 7;

select unique1, unique2, generate_series(1,10)
  from tenk1 order by tenthous limit 7;
 unique1 | unique2 | generate_series 
---------+---------+-----------------
(0 rows)

-- use of random() is to keep planner from folding the expressions together
explain (verbose, costs off)
select generate_series(0,2) as s1, generate_series((random()*.1)::int,2) as s2;

select generate_series(0,2) as s1, generate_series((random()*.1)::int,2) as s2;
 s1 | s2 
----+----
  0 |  0
  1 |  1
  2 |  2
(3 rows)

explain (verbose, costs off)
select generate_series(0,2) as s1, generate_series((random()*.1)::int,2) as s2
order by s2 desc;

select generate_series(0,2) as s1, generate_series((random()*.1)::int,2) as s2
order by s2 desc;
 s1 | s2 
----+----
  2 |  2
  1 |  1
  0 |  0
(3 rows)

-- test for failure to set all aggregates' aggtranstype
explain (verbose, costs off)
select sum(tenthous) as s1, sum(tenthous) + random()*0 as s2
  from tenk1 group by thousand order by thousand limit 3;

select sum(tenthous) as s1, sum(tenthous) + random()*0 as s2
  from tenk1 group by thousand order by thousand limit 3;
 s1 | s2 
----+----
(0 rows)

